scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -235,6 +235,8 @@ def _create_command(self, robot_test):
             command = [self.generator_cmd, "run_renode_test"]
         elif self.call_make_run:
             command = [self.generator_cmd, "run"]
+        elif self.instance.testsuite.type == "unit":
+            command = [self.binary]
         else:
             binary = os.path.join(self.get_default_domain_build_dir(), "zephyr", "zephyr.exe")
```

Comment:
Old code was having special treatment for `native`, like this:
https://github.com/zephyrproject-rtos/zephyr/blob/11de74c11f6ccdea9875c54c722bc0c29e8023d5/scripts/pylib/twister/twisterlib/handlers.py#L328-L330
so wouldn't it be better to keep the test for native and not introduce another test  (`unit`)?

(unsure who passes the `type_str` value in old code, and whether that information is available in `_create_command()`)

Reply:
It is quite tricky. I have a talk with @gchwier. Setting of binary path had to be moved to _create_command() because domains.yaml file is not available when SimulatorHandler is created (_init_, this is where the logic used to be). Those "Handlers" have a tricky logic, since the base class (BinaryHandler) is loaded with logic for cases depended on child classes instead having such differences in child classes themselves. We preferred not to refactor it to not introduce more errors.